src/commands/installCosmosDBExtension.ts

```diff
@@ -3,17 +3,20 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { extensions } from 'vscode';
 import { AzureTreeItem, IActionContext } from 'vscode-azureextensionui';
 import { CosmosDBTreeItem } from '../explorer/CosmosDBTreeItem';
 import { installExtension } from '../utils/installExtension';
 
 export async function installCosmosDBExtension(context: IActionContext, treeItem: AzureTreeItem): Promise<void> {
     const extensionId: string = 'ms-azuretools.vscode-cosmosdb';
-    if (await installExtension(extensionId)) {
-        if (treeItem.parent) {
-            await treeItem.parent.refresh();
-            if ((<CosmosDBTreeItem>treeItem.parent).cosmosDBExtension) {
-                context.telemetry.properties.installedCosmos = 'true';
+    if (!extensions.getExtension(extensionId)) {
```

Comment:
You don't actually need to do this here since the treeItem doesn't appear unless the extension isn't installed: https://github.com/microsoft/vscode-azureappservice/blob/master/src/explorer/CosmosDBTreeItem.ts#L50